analytics-api/src/main/java/org/exoplatform/analytics/model/filter/AnalyticsPercentageFilter.java

```diff
@@ -216,7 +216,7 @@ private AnalyticsAggregation getXAxisAggregation() {
     if (customPeriod != null) {
       long diffInDays = customPeriod.getDiffInDays();
       if (diffInDays > 0) {
-        long offset = (xAxisAggregation.getMinBound() / 86400000l) % diffInDays;
+        long offset = (xAxisAggregation.getMinBound() / 86400l) % diffInDays;
```

Comment:
I see that xAxisAggregation.getMinBound() returns milliseconds, why do you devide by 1000 ?

Reply:
After analyzing and understanding how elasticsearch using the offset parameter in the date_histograms as explained in the elastic official docs : https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html#search-aggregations-bucket-datehistogram-offset

The offset in the case of a custom period (means a selected interval of days from calendar) should be the default value 0h to allow each bucket to start from midnight to midnight.
example : 
select custom period from 4 November to 5  ===>
current period will be 4 -> 5
last period will be 2 -> 3

The **interval** param in the date_histogram will be **2days**
The **offset** will be 0h (the default value)

Result two buckets 
1 - Bucket of two days from 2 -> 3 starts from midnight to midnight which is the number of connected users in the last period.
2 - Bucket of two days from 4 -> 5 starts from midnight to midnight  which is the number of connected users of current selected period .

**By this way i get the percentage of the current period and the diff compared with the last period**

Reply:
You must think of offset as the number of hours of User TimeZone. So midnight to midnight for FR user is not the same as midnight to midnight for user located in US.

Reply:
You must keep in mind that testing is very important to be make sure to have this working well.
Anyway, the offset in days was set to be sure to not have wrong results since the Xdays interval (use 4days interval for example) will rely on number of days since 1st of January 1970. So if you want to use 4 days interval histogram results since `12 Nov. 2021`, you will have to set offset to `3d`. If you select `11 Nov. 2021` as start date, you will have to set offset to `2d`. In fact, by choosing 4d interval histogram, it's like ES will divide all dates since `1st of January 1970` by 4days interval, so to make sure to have 4days interval starting from the desired day, you will have to make a transition with the offset parameter. (`time_zone` parameter is managed elsewhere)